core/bash_impl.py

```diff
@@ -300,6 +300,27 @@ def SparseArray_SetElement(sparse_val, index, s):
     return 0
 
 
+def SparseArray_UnsetElement(sparse_val, index):
+    # type: (value.SparseArray, mops.BigInt) -> int
+    """A non-zero return value represents an error code. If this
+    function returns 1, the specified index is out of range.
+
+    """
+
+    index, ok = _SparseArray_CanonicalizeIndex(sparse_val, index)
+    if not ok:
```

Comment:
It might be better to use integer error codes for both, rather than bool, and int

Also I think it would be nice to have

`if error == bash_impl.IndexOutOfRange`

instead of hard-coding `1`  .    Then we don't need the comment, and it is more readable

Reply:
Yeah, so we go back to the discussion at https://github.com/oils-for-unix/oils/pull/2157#issue-2708501765. I'm happy to refactor them using named constants.

There are already many codes using the current approach with `int`, so I'd rather do it globally in a separate PR than including the global refactoring in this PR (which is for `unset -v sp[i]`).

Reply:
I fixed it and rebased the PR on top of the latest master.

Reply:
I created PR #2175.